extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -630,7 +672,7 @@ export async function newDeployment(
   // Step: New Credentials - Enter the API Key
   // ***************************************************************
   async function inputAPIKey(input: MultiStepInput, state: MultiStepState) {
-    if (newCredentialByAnyMeans()) {
+    if (newCredentialByAnyMeans() && serverType !== ServerType.SNOWFLAKE) {
```

Comment:
Same observation here as in `newCredential`, can this checks be done earlier and determine which method from `inputAPIKey` or `inputSnowflakeConnection` should be called. You could also benefit from returning early, instead of having big `if` blocks.